src/devices/DCMotor/DCMotor.cs

```diff
@@ -41,58 +41,130 @@ public virtual void Dispose()
+        public static DCMotor Create(PwmChannel speedControlChannel)
         {
-            if (settings.UseEnableAsPwm)
-            {
-                if (settings.Pin0.HasValue && settings.Pin1.HasValue && settings.PwmChannel != null)
-                {
-                    return new DCMotor3Pin(
-                        settings.PwmChannel,
-                        settings.Pin0.Value,
-                        settings.Pin1.Value,
-                        settings.Controller);
-                }
-
-                throw new ArgumentException("When enable pin is used for PWM all pins and PwmChannel must be set.");
-            }
-            else
-            {
-                int? pin0 = settings.Pin0;
-                int? pin1 = settings.Pin1;
-
-                if (!pin0.HasValue && pin1.HasValue)
-                {
-                    pin0 = pin1;
-                    pin1 = null;
-                }
-
-                if (settings.PwmChannel != null)
-                {
-                    if (pin0.HasValue == pin1.HasValue)
-                    {
-                        throw new ArgumentException("Only one of Pin0 and Pin1 must be set when PWM is specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(
-                        settings.PwmChannel,
-                        pin0,
-                        settings.Controller);
-                }
-                else
-                {
-                    if (!pin0.HasValue)
-                    {
-                        // pin1 must be null now since we swap values earlier when pin0 == null
-                        throw new ArgumentException("Pin0 or Pin1 must be set when PWM is not specified.");
-                    }
-
-                    return new DCMotor2PinNoEnable(new SoftwarePwmChannel(pin0.Value, 50, 0.0, controller: settings.Controller), pin1, settings.Controller);
-                }
-            }
+            if (speedControlChannel == null)
+                throw new ArgumentNullException(nameof(speedControlChannel));
+
+            return new DCMotor2PinNoEnable(speedControlChannel, -1, null);
+        }
+
+        /// <summary>
+        /// Creates DCMotor instance which allows to control speed in one direction.
+        /// </summary>
+        /// <param name="speedControlPin">Pin used to control the speed of the motor with software PWM</param>
+        /// <param name="controller">GPIO controller related to the pin</param>
+        /// <returns>DCMotor instance</returns>
+        public static DCMotor Create(int speedControlPin, GpioController controller = null)
+        {
+            if (speedControlPin == -1)
+                throw new ArgumentOutOfRangeException(nameof(speedControlPin));
+
+            controller = controller ?? new GpioController();
+            return new DCMotor2PinNoEnable(
+                new SoftwarePwmChannel(speedControlPin, 50, 0.0, controller: controller),
```

Comment:
This is a bit confusing as well, so you are creating a DCMotor that is driven by software pwm in a specified controller, but then you are also passing that controller over to the DCMotor base. This means you could have the case where that guy is null, and you initialize two difierent instances of GpioControllers, one in the soft pwm and one in the dcmotor base class constructor.

Reply:
that's not possible because of `controller = controller ?? new GpioController();` line above

Reply:
it is possible, because inside the SoftwarePwm channel you have the exact same line, where it will also initialize a controller if it gets passed a null, so you end up with 2 GpioControllers

Reply:
that's not the case in this case because local `controller` is not null after `controller = controller ?? new GpioController()` and both DCMotor2PinNoEnable and SoftwarePwmChannel will get that non-null value. The static Create doesn't call base class. I'll validate remaining Create methods to make sure that is not the case as well

Reply:
Seems this was a case for:

```csharp
public static DCMotor Create(PwmChannel speedControlChannel, int directionPin, int otherDirectionPin, GpioController controller = null)
```

so possibly you commented under wrong overload